gadgets/mri_core/CoilReductionGadget.cpp

```diff
@@ -1,122 +1,86 @@
-/*
-* CoilReductionGadget.cpp
-*
-*  Created on: Dec 5, 2011
-*      Author: hansenms
-*/
-
 #include "CoilReductionGadget.h"
-#include <boost/algorithm/string.hpp>
-#include <boost/algorithm/string/split.hpp>
-#include "ismrmrd/xml.h"
 
-namespace Gadgetron{
+namespace Gadgetron {
 
-    CoilReductionGadget::CoilReductionGadget() {
-    }
+CoilReductionGadget::CoilReductionGadget(const Core::Context& context, const Core::GadgetProperties& props)
+    : Core::ChannelGadget<Core::Acquisition>(context, props) {
+    auto h = (context.header);
+    coils_in_ = h.acquisitionSystemInformation->receiverChannels ? *h.acquisitionSystemInformation->receiverChannels : 128;
 
-    CoilReductionGadget::~CoilReductionGadget() {
+    std::string coil_mask_int = coil_mask;
+
+    if (coil_mask_int.compare(std::string("")) == 0) {
+        if (coils_out <= 0) {
+            GDEBUG("Invalid number of output coils %d\n", coils_out);
+            // TODO: How to throw Gadget failures?
```

Comment:
Why can you not just throw a runtime_error?

Reply:
I just realized many of these TODO comments are from when I first started this conversion process a few months back and was less familiar with the Gadgetron log/error approaches - at least they're doing their job haha. Updating to using exception